### PR TITLE
add type specs to shell_docs

### LINKED ISSUE #8544: Add spec to shell docs
**Is your feature request related to a problem? Please describe.**

For markdown docs the spec is currently not displayed for shell docs:

```
 foldl(Fun, Acc0, List)

  Calls Fun(Elem, AccIn) on successive elements A of List,
  starting with AccIn == Acc0. Fun/2 must return a new
```

**Describe the solution you'd like**

The spec should be show as it does in 26.2

```
  -spec foldl(Fun, Acc0, List) -> Acc1
                 when
                     Fun :: fun((Elem :: T, AccIn) -> AccOut),
                     Acc0 :: term(),
                     Acc1 :: term(),
                     AccIn :: term(),
                     AccOut :: term(),
                     List :: [T],
                     T :: term().

  Calls Fun(Elem, AccIn) on successive elements A of List, starting with AccIn == Acc0. Fun/2 must return

```

### PR DESCRIPTION
**Pull request by kikofernandez**
adds type specifications to `shell_docs.erl`. This allows users to type `ht(ssh, daemon)` / `h(erlang, spawn_link)` and get the documentation together with the spec.

The types shown are those included in the `beam` files. Systems compiled without `debug_info` will not have the documentation in the shell.

One test case in `shell_docs_SUITE.erl` has been parallelised due to the amount of time taken to render the new documentation.

Closes #8544

### CODE REVIEW

File: lib/stdlib/src/shell_docs.erl [hunk 1]
```diff
@@ -539,14 +535,30 @@ render(_Module, Function, Arity, #docs_v1{ } = D) ->
...
-                   end, Docs), DocV1, Config).
+render(Module, Function, Arity, #docs_v1{ }=DocV1, Config)
+  when is_atom(Module), is_atom(Function), is_integer(Arity),  is_map(Config) ->
+    render_fn(Module, Function, Arity, DocV1, Config).
+
+%% this function resembles render/5 except that the type allows
+%% 'Arity :: arity() | none()'. The reason for doing this is to
+%% not change existing type specs and to cascade calls from render/2, rennder/3,
+%% and render/4 into a general function that performs all the work.
+-doc false.
```
Comment (garazdawi) [inline lib/stdlib/src/shell_docs.erl:546]:
```suggestion
```
As this function is not exported, there is no need for `-doc false.`

File: lib/stdlib/src/shell_docs.erl [hunk 2]
```diff
@@ -575,7 +582,65 @@ get_type_doc(Module, Type, Arity) ->
...
+    {error, _} -> #{};
+    non_existing -> #{};
+    {error,beam_lib,{file_error,_,enoent}} -> #{}
+  end.
+
+find_path(Module) ->
+    case code:which(Module) of
+        preloaded ->
+            PreloadedPath = filename:join(code:lib_dir(erts),"ebin"),
+            PreloadedPath ++ "/" ++ atom_to_list(Module);
```
Comment (garazdawi) [inline lib/stdlib/src/shell_docs.erl:613]:
```suggestion
            filename:join(PreloadedPath, atom_to_list(Module) ++ ".beam");
```
Or windows will/may break. Although not required by the current `beam_lib:chunks/2` API, I think it is prudent to add the ".beam" extension.

**Review by garazdawi — APPROVED** [review]
Some small comments, but otherwise it looks good to me.

File: lib/stdlib/src/shell_docs.erl [hunk 3]
```diff
@@ -575,7 +581,66 @@ get_type_doc(Module, Type, Arity) ->
...
+    %% to existing EEP-48 {**Kind**, Name, Arity} format, where Kind
+    %% ranges over these values. This is needed to differentiate
+    %% function, callback, and types when looking up their type specs
+    Acc = #{type => #{}, 'function' => #{}, 'callback' => #{}},
+    lists:foldl(fun filter_exported_types/2, Acc, AST)
+  else
+    false -> #{};
+    {error, _} -> #{};
+    non_existing -> #{};
+    {error,beam_lib,{file_error,_,_}} -> #{}
```
Comment (garazdawi) [inline lib/stdlib/src/shell_docs.erl:606]: Can you add some comments about which ?= can return which error?

**Comment by kikofernandez** [comment]
This PR times out on some lab machines, checking what is going on.

**Pull request merged**